tests/foreman/ui/test_host.py

```diff
@@ -2404,3 +2405,159 @@ def test_all_hosts_bulk_delete(target_sat, function_org, function_location, new_
+    creates a new activation key for the CV in the lce,
+    registers the RHEL content host with the activation key,
+    updates the capsule's taxonomies
+    adds the lifecycle environment to the capsule's content.
+
+    Fixture returns module_target_sat, org, lce, capsule, content_view, loc, ak
+    """
+    product_name, lce_name = (gen_string('alpha') for _ in range(2))
+    repos_to_enable = ['rhae2.9_el8']
+
+    org = module_sca_manifest_org
+    loc = module_location
+
+    product = module_target_sat.api.Product(
+        name=product_name,
+        organization=org.id,
+    ).create()
+
+    repository = module_target_sat.api.Repository(
+        product=product,
+        content_type=REPO_TYPE['file'],
+        url=CUSTOM_FILE_REPO,
+    ).create()
+
+    for repo in repos_to_enable:
+        module_target_sat.cli.RepositorySet.enable(
+            {
+                'organization-id': org.id,
+                'name': constants.REPOS[repo]['reposet'],
+                'product': constants.REPOS[repo]['product'],
+                'releasever': constants.REPOS[repo]['version'],
+                'basearch': constants.DEFAULT_ARCHITECTURE,
+            }
+        )
+
+    lce = module_target_sat.cli_factory.make_lifecycle_environment(
+        {'name': lce_name, 'organization-id': org.id}
+    )
+
+    # Create CV
+    content_view = module_target_sat.api.ContentView(organization=org.id).create()
+    # Add repos to CV
+    content_view.repository = [repository]
+    content_view = content_view.update(['repository'])
+    # Publish that CV and promote it
+    content_view.publish()
+    content_view.read().version[0].promote(data={'environment_ids': lce.id})
+
+    ak = module_target_sat.api.ActivationKey(
+        content_view=content_view, organization=org.id, environment=lce.id
+    ).create()
+
+    # Edit capsule's taxonomies
+    capsule = module_target_sat.cli.Capsule.update(
+        {
+            'name': module_capsule_configured.hostname,
+            'organization-ids': org.id,
+            'location-ids': loc.id,
+        }
+    )
+
+    module_target_sat.cli.Capsule.content_add_lifecycle_environment(
+        {
+            'id': module_capsule_configured.nailgun_capsule.id,
+            'organization-id': org.id,
+            'lifecycle-environment': lce.name,
+        }
+    )
+
+    return module_target_sat, org, lce, capsule, content_view, loc, ak
+
+
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_match('[78]')
+def test_change_content_source(session, change_content_source_prep, rhel_contenthost):
+    """
+    This test excercises different ways to change host's content source
+
+    :id: 5add68c3-16b1-496d-9b24-f5388013351d
+
+    :expectedresults: Job invocation page should be correctly generated by the change content source action, generated script should also be correct
```

Comment:
I'd break the line here :)

Reply:
sadly test-docs fails if I break that line :(